File: server/internal/products/product.go
package products

type Product struct {
	SKU      string  `json:"sku"`
	Title    string  `json:"title"`
	Price    float64 `json:"price"`
	Currency string  `json:"currency"`
	Stock    int     `json:"stock"`
	Rating   float64 `json:"rating"`
	Image    string  `json:"img"`
}

type Category struct {
	ID     int    `json:"id"`
	Name   string `json:"name"`
	Slug   string `json:"slug"`
	Parent int    `json:"parent_id"`
}

File: app/lib/models/product.dart
class Product {
  final String sku;
  final String title;
  final double price;
  final String currency;
  final int stock;
  final double rating;
  final String image;

  Product({
    required this.sku,
    required this.title,
    required this.price,
    required this.currency,
    required this.stock,
    required this.rating,
    this.image = "img/default.jpg"
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      sku: json['sku'] ?? '',
      title: json['title'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      currency: json['currency'] ?? '',
      stock: json['stock'] ?? 0,
      rating: (json['rating'] ?? 0).toDouble(),
      image: json['img'] ?? '',
    );
  }
}
